### ISSUE TITLE
Travis plan

### ISSUE THREAD

**Issue by wvangeit** [lbologna/bsp_data_repository#31]
Could I propose we create a travis plan to check the zip files every time they are updated ?

That way we create some 'memory' about passed issues, and make sure they don't reappear.

Things I can think of checking:
- are all the json files of the correct format
- do the models / morphologies load in Neuron
- are all the MOD files from the same release
- ...

This is not wasted effort once things move into Nexus, because we need exactly the same check before importing into nexus.
So we should make it in a way that it is easily usable outside of travis.

@lbologna did you ever create a Travis plan ? Could you attach one to this repo ? Once that is done I can create some code to do initial checks and we can work from there.

**Comment by wvangeit** [comment]
@lbologna could you please answer my question about the travis plan, we urgently need one. Issues like https://github.com/lbologna/bsp_data_repository/issues/34 could easily be automatically detected once we have such a plan.
Thanks.

**Comment by lbologna** [comment]
Hello Werner,

sorry for late reply.
I never created a travis plan and will not be able to work on that for the next weeks.
We will add a few optimizations in the following weeks that we can check manually for the moment.

ps. the migration to the container should come in the following weeks as well, will the travis plan work for that new location as it is created here?

**Comment by wvangeit** [comment]
I don't know about the details of that container. If it's stored outside of github I guess it might be difficult.
But the main thing is that we have:
- a script that checks the consistency of the files
- a system that automatically runs this script every time the files are updated

**Comment by wvangeit** [comment]
@clupascu For the neuron tests. For now I would just instantiate the cell with the morph it was optimized for and run a little simulation with it. Would already be enough to catch obvious errors.
In a next step you could rerun the optimization protocols on it, but that's more complicated I guess.

**Comment by clupascu** [comment]
@wvangeit We will try to do this, even if for now we are still having problems with loading a hoc template. It gives segmentation fault error with pytest even outside Travis with one of the latest neuron github version. We are still testing this. Thanks.

**Comment by wvangeit** [comment]
@clupascu could you post a link to the travis website of the plan ? I can't find it on travis-ci.org. 
As a suggestion, it would be useful to add a badge on the main page of this repo that points to the travis build status (https://docs.travis-ci.com/user/status-images/).

**Comment by clupascu** [comment]
@wvangeit The link of the travis website of the plan is https://travis-ci.com/lbologna/bsp_data_repository.
I added a badge to the travis build status as you suggested. Please check. Thanks

**Comment by wvangeit** [comment]
Ok, looks good !

Two more suggestions:
- if you add a cache in travis, it won't have to rebuild neuron all the time, you can specify which directories to cache: https://github.com/BlueBrain/BluePyOpt/blob/master/.travis.yml#L15
- I would make the pytest more verbose so you can follow better the tests. Maybe there is some '-v' option ?

**Comment by clupascu** [comment]
Thank you again for your suggestions. I added a cache in travis and I made the pytest more verbose with '-v' option.

**Comment by clupascu** [comment]
@wvangeit can we close this issue?

**Comment by wvangeit** [comment]
Yes, we can close it I think.

**Comment by wvangeit** [comment]
@clupascu I saw that the Travis plan is failing for already a couple of months.
Can this easily be fixed ?

**Comment by clupascu** [comment]
@wvangeit A couple of months ago I tried to test within pytest if modules are imported correctly. This test failed. It seems difficult to import modules in pytest. Do you have any idea on how to do it? In the meanwhile I removed those tests. It shouldn't fail anymore.

**Comment by wvangeit** [comment]
Ok. Could I suggest you make changes to the repo using pull requests and not commit directly into the master ? That way travis will check the PR for you before it gets merged into the master.

About importing the modules ? Where are these modules located ? Are they part of the repo ?

**Comment by clupascu** [comment]
Ok. The modules are part of the repo.

**Comment by wvangeit** [comment]
Maybe there is some solution in this ticket ?
https://stackoverflow.com/questions/10253826/path-issue-with-pytest-importerror-no-module-named-yadayadayada
